### PR TITLE
Crossheadings and attachment eIds

### PR DESCRIPTION
**Pull request by longhotsummer**
Implements the data migration parts of #1347

### CODE REVIEW

File: indigo_api/data_migrations/__init__.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        self.ns = document.doc.namespace
+        if self.migrate_xml(document.doc.root):
+            return True
+
+    def migrate_xml(self, xml):
+        changed = False
+        self.eid_mappings = {}
+
+        for att in xml.xpath('//a:attachment', namespaces={'a': self.ns}):
+            changed = changed or self.migrate_element(att)
```
Comment (goose-life) [inline indigo_api/data_migrations/__init__.py:66]: does this mean only the first att will have migrate_element() called on it?

File: indigo_api/migrations/0009_migrate_attachment_eids.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                # update sec_1 -> att_1__sec_1
+                Annotation.objects.using(db_alias)\
+                    .filter(document_id=doc.id, anchor_id=old)\
+                    .update(anchor_id=new)
+
+                # update prefixed anchors: att_1/sec_1 -> att_1__sec_1
+                prefix = new.split('__', 1)[0] + '/'
+                assert(prefix.startswith('att_'))
+                old = prefix + old
+                new = prefix + new
```
Comment (goose-life) [inline indigo_api/migrations/0009_migrate_attachment_eids.py:31]: prefix ends in /, and we don't want that in `new`, right?

File: indigo_api/migrations/0009_migrate_attachment_eids.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    dependencies = [
+        ('indigo_api', '0008_taxonomy_vocabulary'),
+    ]
+
+    operations = [
+        migrations.RunPython(forwards, migrations.RunPython.noop),
+
+        # strip att_1/ from att_1/att_1__sec_1 in anchor ids
+        migrations.RunSQL(
```
Comment (goose-life) [inline indigo_api/migrations/0009_migrate_attachment_eids.py:47]: just out of interest, why do we need this if we're updating the Annotation objects above?

File: indigo_api/data_migrations/__init__.py (hunk 1 above)
Comment (longhotsummer) [reply indigo_api/data_migrations/__init__.py]: Corrected.

File: indigo_api/migrations/0009_migrate_attachment_eids.py (hunk 2 above)
Comment (longhotsummer) [reply indigo_api/migrations/0009_migrate_attachment_eids.py:32]: I've updated the comment. We preserve the `att_1/att_1__foo` here, and then strip them everywhere with SQL in the next step.

The SQL ensures that we update all annotations, not just those for re-written documents.

File: indigo_api/migrations/0009_migrate_attachment_eids.py (hunk 3 above)
Comment (longhotsummer) [reply indigo_api/migrations/0009_migrate_attachment_eids.py:50]: The above code keeps the `att_1/` prefix. This code then strips it from everything.

**Review by goose-life — APPROVED** [review]

**Pull request merged**